routes/api/users.js

```diff
@@ -0,0 +1,148 @@
+import express from 'express';
+import dotenv from 'dotenv';
+
+import { auth } from '../../config/config-passport.js';
+import { uploadImage } from '../../config/config-multer.js';
+
+dotenv.config();
+
+export const usersRouterFunction = usersService => {
+  const { logOutUser, getUser, addUser, loginUser, patchUser, patchAvatar } = usersService;
+  const usersRouter = express.Router();
+
+  usersRouter.get('/current', auth, async (req, res, next) => {
+    const { id: userId } = req.user;
+    try {
+      const user = await getUser(userId);
+      if (!user) {
+        return res.status(404).json(`Error! User not found!`);
+      }
+      const { email, subscription } = user;
+      return res.status(200).json({
+        status: 'success',
+        code: 200,
+        data: { email, subscription },
+      });
+    } catch (err) {
+      res.status(500).json(`An error occurred while getting the contact: ${err}`);
+    }
+  });
+
+  usersRouter.post('/signup', async (req, res, next) => {
+    const { body } = req;
+
+    if (!('email' in body) || !('password' in body)) {
+      return res.status(400).json('Error! Missing password or email field!');
+    }
+
+    try {
+      const user = await addUser(body);
+      if (user === 409) {
+        return res.status(409).json({ message: 'Email in use' });
+      }
+      const { email, subscription } = user;
+      return res.status(201).json({
+        status: 'success',
+        code: 201,
+        user: { email, subscription },
+      });
+    } catch (err) {
+      res.status(500).json(`An error occurred while adding the user: ${err}`);
+    }
+  });
+
+  usersRouter.post('/login', async (req, res, next) => {
+    const { body } = req;
+
+    if (!('email' in body) || !('password' in body)) {
```

Comment:
use joi validation instead of this if. Its easier to maintain also you can add more validation criteria like string `length` etc.